llvm/lib/Target/SPIRV/SPIRVEmitIntrinsics.cpp

```diff
@@ -597,7 +636,7 @@ void SPIRVEmitIntrinsics::insertAssignTypeIntrs(Instruction *I) {
       if (isa<UndefValue>(Op) && Op->getType()->isAggregateType())
         buildIntrWithMD(Intrinsic::spv_assign_type, {IRB->getInt32Ty()}, Op,
                         UndefValue::get(IRB->getInt32Ty()), {});
-      else
+      else if (!isa<Instruction>(Op)) // TODO: This case could be removed
```

Comment:
I will remove/refactor this in the next patch, so that it does not obscure the changes here